### PR TITLE
Update API urls to avoid redirects

### LINKED ISSUE #556: Use new API URLs in requests to avoid redirects
## Problem

The API URL structure changed a little bit when the regular URL conf was replaced with DRF router. Now, there are trailing slashes, and the requests from the frontend are redirected using 301.

## Description

It would be cleaner to add the trailing slashes to the requests by the frontend to avoid redirects.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #556 by @obulat

## Description

This PR adds trailing slashes to the old request URLs to avoid 301 redirects.

```
https://api.openverse.engineering/v1/images?q=cat[HTTP/2 301 Moved Permanently 579ms]
https://api.openverse.engineering/v1/images/?q=cat [HTTP/2 200 OK 288ms]
```
These changes are added to the `search`, 'single result pages' for image and audio, and the `thumbs` endpoints.

## Testing Instructions

Run the app using `pnpm run dev`, and try searching or looking at the single result pages. You should not see any 301 redirects in the console.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by krysal — COMMENTED** [review]
Great! This is critical for `report/` requests. Could we include them here? Ty!

**Comment by obulat** [comment]
Thank you for spotting this, @krysal ! I haven't thought about other requests at all :(

**Comment by sarayourfriend** [comment]
@obulat I wonder if the ApiService shouldn't enforce this somehow? We could do this using an [axios interceptor](https://axios-http.com/docs/interceptors) for example.

**Comment by obulat** [comment]
> @obulat I wonder if the ApiService shouldn't enforce this somehow? We could do this using an [axios interceptor](https://axios-http.com/docs/interceptors) for example.

It's a nice idea to use one place to enforce the correct URL usage, but I'm not sure it can be used here. The search URLs (https://api.openverse.engineering/v1/images/?q=cat) doesn't need a trailing slash, and all the others do.

**Comment by sarayourfriend** [comment]
Yeah I guess we would have to split the URL parts, append the trailing slash if it's missing, then put it back together? 🤔

**Review by dhruvkb — APPROVED** [review]
I like that we have the trailing slash appended to all the URLs in use (and those that don't will not fail, just redirect). Enforcing this in the ApiServices seems excessive to me, especially because it shows clear warnings in the console (that can be easily fixed) and does not break anything.

**Review by krysal — APPROVED** [review]

File: src/data/audio-service.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const AudioService = {
 
     return dev
       ? Promise.resolve({ data: sampleAudioResponses.detail })
-      : ApiService.get('audio', params.id)
+      : ApiService.get('audio/', params.id)
```
Comment (zackkrida) [inline src/data/audio-service.js:42]: Won't this result in double slashes? The `get` function in the api service already adds a slash after the resource

Comment (sarayourfriend) [reply src/data/audio-service.js:42]: @dhruvkb automating it via some axios adapter would prevent confusions like this ^. That's the main benefit to me, then we never have to think about it again. I see it the same as when we adopt an ESLint rule, just in this case we can fix the problem easily at runtime.

**Comment by obulat** [comment]
> Yeah I guess we would have to split the URL parts, append the trailing slash if it's missing, then put it back together? 🤔

It turned out to be simpler than that, as the axios request already splits the url into parts, and it is the`config.url` param that needs to have a trailing slash. Thank you for suggesting validation, @sarayourfriend !

**Pull request merged**